lib/streamlit/elements/form.py

```diff
@@ -110,15 +110,60 @@ def _build_duplicate_form_message(user_key: Optional[str] = None) -> str:
 
 class FormMixin:
     def form(self, key: str):
-        """TODO
+        """Create a form that batches elements together with a "Submit" button.
 
-        Parameters
-        ----------
-        key
+        A form is a container that visually groups other elements and
+        widgets together, and contains a Submit button. When the form's
+        Submit button is pressed, all widget values inside the form will be
+        sent to Streamlit in a batch.
 
-        Returns
-        -------
+        To add elements to the returned form object, you can use "with" notation
+        (preferred) or just call methods directly on the returned object. See
+        examples below.
+
+        Forms have a few constraints:
+        - Every form must contain a `st.form_submit_button`.
+        - You cannot add a normal `st.button` to a form.
+        - Forms can appear anywhere in your app (sidebar, columns, etc),
+          but they cannot be embedded inside other forms.
 
+        Parameters
+        ----------
+        key : str
+            A string that identifies the form. Each form must have its own
+            key. (This key is not displayed to the user in the interface.)
```

Comment:
Minor: I noticed that in other API docs(for e.g. [here](https://docs.streamlit.io/en/stable/api.html?highlight=container#streamlit.radio)) we use the definition: 

> key (str) – An optional string to use as the unique key for the widget. If this is omitted, a key will be generated for the widget based on its content. Multiple widgets of the same type may not share the same key.

Can we use the above for consistency?

Reply:
The key is **not** optional.
> key (str) - A string to use as the unique key for the form. Multiple forms may not share the same key.

Reply:
Ah, good catch @kantuni

@tconkling : please ignore my previous comment.